File: simulation.jl
using ODE;

include("./model/setParamConst.jl");
include("./model/setVarEnum.jl");
include("./model/diffeq.jl");
include("./model/initialValues.jl");

p = setParamConst();
u0 = initialValues();

t = 0.0:1.0:1800.0;
condition = 8;

ERK_act = zeros(length(t),condition);
Akt_act = zeros(length(t),condition);

for i=1:condition
    if i==1
        u0[E] = 0.0;
        u0[H] = 0.5;
    elseif i==2
        u0[E] = 0.0;
        u0[H] = 10.0;
    elseif i==3
        u0[E] = 0.5;
        u0[H] = 0.0;
    elseif i==4
        u0[E] = 0.5;
        u0[H] = 0.5;
    elseif i==5
        u0[E] = 0.5;
        u0[H] = 10.0;
    elseif i==6
        u0[E] = 10.0;
        u0[H] = 0.0;
    elseif i==7
        u0[E] = 10.0;
        u0[H] = 0.5;
    elseif i==8
        u0[E] = 10.0;
        u0[H] = 10.0;
    end

    (sol_t,sol_u) = ode23s(diffeq,u0,t;points=:specified);

    for j=1:length(t)
        ERK_act[j,i] = sol_u[j][ERKstar] + sol_u[j][pERK_ERKPpase];
        Akt_act[j,i] = sol_u[j][Aktstar];
    end

end

File: simulation.py
import numpy as np
from scipy.integrate import ode

from model.name2idx import parameters as C
from model.name2idx import variables as V
from model.param_const import f_params
from model.initial_condition import initial_values
from model.differential_equation import diffeq


def solveode(diffeq,y0,tspan,args):
    sol = ode(diffeq)
    sol.set_integrator('vode',method='bdf',min_step=1e-8,with_jacobian=True)
    sol.set_initial_value(y0,tspan[0])
    sol.set_f_params(args)

    T = [tspan[0]]
    Y = [y0]

    while sol.successful() and sol.t < tspan[-1]:
        sol.integrate(sol.t+1.)
        T.append(sol.t)
        Y.append(sol.y)

    return np.array(T),np.array(Y)


class Simulation(object):

    tspan = range(1801)
    condition = 8
    
    t = np.array(tspan)
        
    ERK_act = np.empty((len(t),condition))
    Akt_act = np.empty((len(t),condition))
    
    x = f_params()
    y0 = initial_values()
    
    for i in range(condition):
        if i==0:
            y0[V.E] = 0.0
            y0[V.H] = 0.5
        elif i==1:
            y0[V.E] = 0.0
            y0[V.H] = 10.0
        elif i==2:
            y0[V.E] = 0.5
            y0[V.H] = 0.0
        elif i==3:
            y0[V.E] = 0.5
            y0[V.H] = 0.5
        elif i==4:
            y0[V.E] = 0.5
            y0[V.H] = 10.0
        elif i==5:
            y0[V.E] = 10.0
            y0[V.H] = 0.0
        elif i==6:
            y0[V.E] = 10.0
            y0[V.H] = 0.5
        elif i==7:
            y0[V.E] = 10.0
            y0[V.H] = 10.0
        
        (T,Y) = solveode(diffeq,y0,tspan,tuple(x))
        
        ERK_act[:,i] = Y[:,V.ERKstar] + Y[:,V.pERK_ERKPpase]
        Akt_act[:,i] = Y[:,V.Aktstar]
            